packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -284,13 +276,37 @@ export class SerializedStateManager {
 		}
 	}
 
+	private tryRefreshSnapshot(): void {
+		if (
+			this._refreshSnapshotP === undefined &&
```

Comment:
should this also block if latestSnapshot is set, as if we haven't been able to set the snapshot for whatever reason, there probably isn't a point to get it again